### ISSUE TITLE
Lambda layer does not compute masked values properly

### ISSUE THREAD

**Issue by Strateus** [tensorflow/tensorflow#40085]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Windows 10
- TensorFlow installed from (source or binary):
conda binary
- TensorFlow version (use command below):
'2.1.0'
- Python version:
3.7

**Describe the current behavior**

Output of the model with lambda subtracting 0.5 from input, and input as [1, -10], with -10 being masked value
```
<tf.Tensor: shape=(1, 2, 1), dtype=float32, numpy=
array([[[ 0.5],
        [-0.5]]], dtype=float32)
```

**Describe the expected behavior**

Output
```
<tf.Tensor: shape=(1, 2, 1), dtype=float32, numpy=
array([[[ 0.5],
        [ 0. ]]], dtype=float32)
```
or
```
<tf.Tensor: shape=(1, 2, 1), dtype=float32, numpy=
array([[[ 0.5],
        [ -10. ]]], dtype=float32)
```

**Standalone code to reproduce the issue**

```
import numpy as np
from tensorflow import keras
inputs_shape = (2, 1)
inputs = keras.layers.Input(shape=inputs_shape)
masked = keras.layers.Masking(mask_value=-10., input_shape=inputs_shape)(inputs)
output = keras.layers.Lambda(lambda x: x - 0.5)(masked)
model = keras.Model(inputs=inputs, outputs=output)
data = np.array([[1, -10]], dtype=np.float32).reshape(-1, 2, 1)
model(data)
```

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.1, 2.2, nightly version(`2.3.0-dev20200602`) and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/5bbd839c334cc24d5b295f95313a375f/untitled948.ipynb).Thanks!

**Comment by Saduf2019** [comment]
I ran the code shared and the issue still persist on tf-nightly[2.4.0-dev20201012], please find the [gist here](https://colab.research.google.com/gist/Saduf2019/aa3f0249313757de0d0d83f25170565a/untitled431.ipynb).

**Comment by jvishnuvardhan** [comment]
@Strateus I think this is intended behavior. If you look at the input to the `lambda` layer is [1,0] and during `lambda x: x - 0.5` we are deducting 0.5 which results in [0.5,-0.5]. Please take a look at the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/18a34178226c88823b346b15191fad3b/untitled431.ipynb). Thanks!

```
model2 = keras.Model(inputs=inputs, outputs=masked)
model2(data)
```

Output is 

```
<tf.Tensor: shape=(1, 2, 1), dtype=float32, numpy=
array([[[ 1.],
        [-0.]]], dtype=float32)>
```

Please verify once and let me know if I am missing anything. Please close the issue if this was resolved for you. Thanks

**Comment by Strateus** [comment]
@jvishnuvardhan your interpretation of masking seems to be different from the official one: _Masking is a way to tell sequence-processing layers that certain timesteps in an input are missing, and thus should be **skipped** when processing the data._

check here: https://www.tensorflow.org/guide/keras/masking_and_padding

Skipping means you **return input value all the time**, not matter what processing you have inside the layer. An alternative would be to return a value of a mask itself (but this is less common).

Lambda layer **should not** deduct anything for the masked values, but it currently does, which is incorrect.

Your gist shows something else: that lambda layer is working without mask. I was not claiming it is not working with masking. I was claiming **masking inside Lambda layer is not working**.

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!

**Comment by Strateus** [comment]
Omg just fix it, it still persists in 2.7
